data-prepper-api/src/main/java/org/opensearch/dataprepper/model/acknowledgements/AcknowledgementSetManager.java

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.model.acknowledgements;
+
+import org.opensearch.dataprepper.model.event.EventHandle;
+import org.opensearch.dataprepper.model.event.Event;
+import java.util.function.Consumer;
+
+public interface AcknowledgementSetManager {
+    AcknowledgementSet create(final Consumer<Boolean> callback, final long TimeoutMilli);
+    void releaseEventReference(final EventHandle eventHandle, boolean success);
```

Comment:
Can these three methods be kept internal to `data-prepper-core`?

Reply:
I do not think so. For example, source plugins need to use `create` and sinks need to use `releaseEventReference`